### DISCUSSION TITLE
Toolbar on MacOS

### DISCUSSION THREAD

**Discussion by anthonyirwin82** [picoe/Eto#2140] [Q&A]
Hi,

I have been looking at eto.forms 2.6.1 and noticed that the toolbar items are not clickable and the command set for the toolbar items does not execute. I am using MacOS 12.2.

If I run the Gtk version on the same computer the toolbar items show up on the left and are clickable and run the commands.

I was wondering why this is?

EDIT: there is a button with no text or icon above the text that you can click on MacOS.

Is there a way to put the text inside the button on mac or add an icon to it?

Thanks,
Anthony

**Comment by cwensley** [comment]
Yes, you need to supply an image for your toolbar items so it works correctly.   In previous versions of macOS they were part of the clickable region of the button but this is no longer the case.  I'm not sure of any way to change that in the current macOS, but I am welcome to any updates to give that as an option.
